packages/server/src/api/controllers/user.ts

```diff
@@ -42,6 +42,7 @@ export async function updateMetadata(
   // this isn't applicable to the user
   delete metadata.roles
   ctx.body = await db.put(metadata)
+  await cache.user.invalidateUser(user._id!)
```

Comment:
This was added during debugging what the problem was, as I thought it could be caching related. It wasn't, but I feel like having this here is correct so I left it in.

Reply:
Ah - the user here is the global metadata user, technically this comes with no negative - but the user metadata should never have any impact on the actual global user itself. The two concepts are quite separate, despite looking very related, but we try to make sure its not possible to update the global user via the app metadata about the user.

Reply:
Ah, so should I remove this?

Reply:
Sorry - didn't see this comment earlier - yeap I would remove this!